protocol/x/rewards/keeper/keeper.go

```diff
@@ -144,8 +145,14 @@ func (k Keeper) AddRewardSharesForFill(
 		totalNetFeeRevSharePpm = value
 	}
 	totalTakerFeeRevShareQuantums := big.NewInt(0)
-	if value, ok := revSharesForFill.FeeSourceToQuoteQuantums[revsharetypes.REV_SHARE_FEE_SOURCE_TAKER_FEE]; ok {
-		totalTakerFeeRevShareQuantums = value
+
+	// taker revshare is not returned if taker volume is greater than Max30dTakerVolumeQuantums
```

Comment:
We aren't using `Max30dRefereeVolumeQuantums` below?

Reply:
We dont need to - since no taker revshares are returned over 50 million - the if condition is never hit for value > 0

Reply:
The trading rewards logic is below:

If taker volume is <50M (equal statement: they are eligible for affiliate rev share, if referred), then the max possible affiliate rev share is deducted from their trading rewards score.

The following implementation doesn't work correctly for a non-referred user under 50M volume